nistats/first_level_model.py

```diff
@@ -557,3 +568,214 @@ def compute_contrast(self, contrast_def, stat_type=None,
+                                 preproc_variant=None, preproc_space=None):
+    """Create FirstLevelModel objects and fit arguments from a BIDS dataset.
+
+    Parameters
+    ----------
+    dataset_path: str
+        Directory of the highest level folder of the BIDS dataset. Should
+        contain subject folders and a derivatives folder.
+
+    task_id: str
+        Task_id as specified in the file names. Of the form _task-*_.
+
+    model_init: dict, optional
+        Is posible to specify the parameters of the FirstLevelModel objects
+        as a dictionary. The same parameter initialization will be applied to
+        all FirstLevelModel objects.
+
+    preproc_variant: str, optional
+        Can specify a particular variant label of the preproc.nii images.
+        As they are specified in the file names.
+
+    preproc_space: str, optional
+        Can specify a particular space label of the preproc.nii images.
+        As they are specified in the file names.
+
+    Returns
+    -------
+    models: list of `FirstLevelModel` objects
+        Each FirstLevelModel object corresponds to a subject. All runs from
+        different sessions are considered together for the same subject to run
+        a fixed effects analysis on them.
+
+    models_fit_kwargs: list of dict
+        For each model there is a corresponding fit_kwargs dictionary. The
+        dictionary contain for certain 'run_imgs' and 'events' keys, and
+        optionally a 'confounds' key. For each run_img there is an events file
+        and a confounds file, so all three lists must have the same size.
+        fit_kwargs are provided as a dictionary to be easily passed to the
+        fit function of the models as model.fit(**kwargs), particularly when
+        no additional modifications of events or confounds is necessary.
+    """
+    # check arguments
+    if not isinstance(dataset_path, str):
+        raise TypeError('dataset_path must be a string, instead %s was given' %
+                        type(task_id))
+    if not os.path.exists(dataset_path):
+        raise ValueError('given path do not exist: %s' % dataset_path)
+    if not isinstance(task_id, str):
+        raise TypeError('task_id must be a string, instead %s was given' %
+                        type(task_id))
+    if model_init is not None and not isinstance(model_init, dict):
+        raise TypeError('model_init must be a dict, instead %s was given' %
+                        type(model_init))
+
+    # check derivatives folder is present
+    derivatives_path = os.path.join(dataset_path, 'derivatives')
+    if not os.path.exists(derivatives_path):
+        raise ValueError('derivatives folder does not exist in given dataset')
+
+    # Get acq specs for models. RepetitionTime and SliceTimingReference.
+    # Only if not given in model_init dictionary
+    # Throw warning if no bold.json is found
+    TR = None
+    SliceTimingRef = 0.0
```

Comment:
I agree, this depends on accepting or not to have model_init as the function argument, if instead we specify all firstlevelmodel parameters this would be there.